py/emitglue.c

```diff
@@ -115,7 +115,7 @@ void mp_emit_glue_assign_native(mp_raw_code_t *rc, mp_raw_code_kind_t kind, cons
     #endif
     #elif MICROPY_EMIT_ARM
     #if (defined(__linux__) && defined(__GNUC__)) || __ARM_ARCH == 7
-    __builtin___clear_cache((void *)fun_data, (uint8_t *)fun_data + fun_len);
+    __builtin___clear_cache((void *)fun_data, (void *)((uint8_t *)fun_data + fun_len));
```

Comment:
```suggestion
    __builtin___clear_cache((void *)fun_data, (char *)fun_data + fun_len);
```

Usually `char *` gets special treatment, so might work instead of the double cast.

Reply:
It worked indeed.  I wonder if there's anything in the spec saying that `char*` can implicitly casted into `void*` whilst `unsigned char*` can't, and why.  The Clang error message was pretty explicit into mentioning "non-void" pointers anyway.

Anyway, thanks for the suggestion, I'll update the PR right away!